src/components/ModularTable/ModularTable.tsx

```diff
@@ -136,6 +140,7 @@ function ModularTable<D extends Record<string, unknown>>({
   getRowId,
   initialSortColumn,
   initialSortDirection,
+  autoResetSortBy = true,
```

Comment:
I wonder if this should be a breaking change and set it to `false`. I'm not sure anyone expects the whole table to be reset when the data changes (this happens even if it's a small nested update to the data) and I can't really think of a situation where you would WANT it to reset. What do you think?